lib/exe/css.php

```diff
@@ -344,6 +360,37 @@ function css_fixreplacementurls($replacements, $location) {
 }
 
 /**
+ * Wrapper for the files, content and mediatype for the event CSS_STYLES_INCLUDED
+ *
+ * @param string $mediatype type ofthe current media files/content set
+ * @param array $files set of files that define the current mediatype
+ * @return array
+ */
+function css_filewrapper($mediatype, $files=array()){
+    return array(
+            'files'                 => $files,
+	        'mediatype'             => $mediatype,
+	        'encapsulate'           => in_array($mediatype, array('screen', 'print')),
```

Comment:
I did not want to dismiss media types ;)
I did not know about the array concept. This could actually work. I'll have to look into it.

I would not create different style.ini files for different contexts as they are a one stop point for all the definitions, right? I think it would make more sense to add more [stylesheet] sections or namespaces, like [stylesheet:foo] (is this possible?).

If we were introducing multiple style.ini files it should work like the plugin types do currently: if there is only one style.ini, it should be in the root of the plugin. If there are more than one context: they should reside in a subfolder (but how should contexts be defined? Fixed list?):

    style
       |-   context1.ini
       |-   context2.ini

Then again: what about files that are reused in different contexts? I'm currently working on a theme with ~50 style files and multiple re-uses. It's a bit tiresome to include them multiple times.

BTW: is it ok to have this lengthy conversation here or should we move to the irc?